# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1751938004 -7200
# Parent  4a9cea9eaa1ff3d4167731e08994681950e94750
Branding: Make './mach build installers-<lang>' work.

diff --git a/toolkit/locales/l10n.mk b/toolkit/locales/l10n.mk
--- a/toolkit/locales/l10n.mk
+++ b/toolkit/locales/l10n.mk
@@ -73,11 +73,14 @@ STAGEDIST = $(ABS_DIST)/l10n-stage/$(MOZ
 endif
 
 include $(MOZILLA_DIR)/toolkit/mozapps/installer/packager.mk
 
+# This will of course only work for Windows. The installer is useless since it don't contain the necessary strings.
+# We're only interested in a localised setup.exe.
 unpack:
 	$(RM) -r -f '$(ABS_DIST)/l10n-stage'
-	$(PYTHON3) $(topsrcdir)/mach --log-no-times artifact install --unfiltered-project-package --distdir '$(ABS_DIST)/l10n-stage/$(MOZ_PKG_DIR)' --verbose
+	mkdir '$(ABS_DIST)/l10n-stage'
+	unzip '$(ABS_DIST)/betterbird-$(MOZ_APP_VERSION).en-US.win64.zip' -d '$(ABS_DIST)/l10n-stage'
 
 # The path to the object dir for the mozilla-central build system,
 # may be overridden if necessary.
 MOZDEPTH ?= $(DEPTH)
@@ -108,8 +111,10 @@ endif
 endif
 	$(NSINSTALL) -D $(DIST)/$(PKG_PATH)
 	mv -f '$(DIST)/l10n-stage/$(PACKAGE)' '$(ZIP_OUT)'
 	if test -f '$(DIST)/l10n-stage/$(PACKAGE).asc'; then mv -f '$(DIST)/l10n-stage/$(PACKAGE).asc' '$(ZIP_OUT).asc'; fi
+	# Repair damage done.
+	echo "en-US" > $(DIST)/../toolkit/locales/default.locale
 
 repackage-zip-%: unpack
 	@$(MAKE) repackage-zip AB_CD=$*
 
